packages/demo/src/index.js

```diff
@@ -162,20 +162,28 @@ async function main() {
   }
 
   async function renderExampleButtons() {
-    const buttonContainer = document.querySelector("#example-streams");
-    ExampleStreams.forEach(async (exampleStream) => {
-
-      const manifestType = await getManifestType(exampleStream.url);
-      const nativelySupportedManifestType = canPlayManifestType(manifestType);
-      const isIOS = /iPhone|iPod/.test(navigator.userAgent);
-      const isIPadOS = /iPad/.test(navigator.userAgent) || (/Macintosh/.test(navigator.userAgent) && "ontouchstart" in document.documentElement);
-      if ((isIOS || isIPadOS) && !nativelySupportedManifestType) return false;
+    const buttonContainer = document.querySelector('#example-streams');
+
+    const playableStreams = await Promise.all(
+      ExampleStreams.map(async (exampleStream) => {
+        const manifestType = await getManifestType(exampleStream.url);
+        const nativelySupportedManifestType = canPlayManifestType(manifestType);
+        const isIOS = /iPhone|iPod/.test(navigator.userAgent);
+        const isIPadOS = /iPad/.test(navigator.userAgent) || (/Macintosh/.test(navigator.userAgent) && 'ontouchstart' in document.documentElement);
```

Comment:
These two may be outside the loop as they should not change

Reply:
In a compiled language I think this would be optimized by the compiler, making readability the greater concern. 

But for JS? Yeah, I should probably move them!